src/attached_probe.cpp

```diff
@@ -964,11 +964,15 @@ void AttachedProbe::attach_kprobe(bool safe_mode)
     return;
   }
 
+  std::string funcname = probe_.attach_point;
+  if (probe_.path.length() > 0)
+    funcname = probe_.path + ":" + funcname;
+
   resolve_offset_kprobe(safe_mode);
   int perf_event_fd = bpf_attach_kprobe(progfd_,
                                         attachtype(probe_.type),
                                         eventname().c_str(),
-                                        probe_.attach_point.c_str(),
+                                        funcname.c_str(),
```

Comment:
The previous PR (that I based mine on) used [this](https://www.kernel.org/doc/Documentation/trace/kprobetrace.rst) as justification for this working, but that link doesn't refer to libbpf. The real question is whether the underlying `perf_event_open` call in [libbpf](https://github.com/iovisor/bcc/blob/d9645f8253d1d8aed0d43503e70dacfb88cb1c72/src/cc/libbpf.c#L1128) is able to take a `module:function` (and properly fail if either doesn't exist).

Reply:
So what happens if the module is some junk string? Is there an eventual failure somewhere else?

Reply:
All right. After digging around in `libbpf` a little deeper, that name just gets passed to `perf_event_open` directly (e.g. it doesn't split the module out from the function name). So, we need to compare bpftrace's behavior versus directly using the kprobe sysfs.

With this PR, `bpftrace` exhibits the following behavior:

```
# Good module, good function
./src/bpftrace -e 'kprobe:kvm:x86_emulate_insn {1}'
[SUCCESS]

# Good module, bad function
./src/bpftrace -e 'kprobe:kvm:nonsense {1}'
[ERROR]

# Bad module, good function
./src/bpftrace -e 'kprobe:nonsense:x86_emulate_insn {1}'
[SUCCESS]

# Bad module, bad function
./src/bpftrace -e 'kprobe:nonsense:nonsense {1}'
[WARNING]
```

Particularly the configuration that shows a bad module *and* a bad function
made me think that this isn't getting passed along to `perf` correctly;
perhaps `bpf_attach_kprobe` doesn't properly handle `module:function`.

*But* if we use the sysfs interface directly, we get the following behavior:

```
# echo 'p:gg kvm:x86_emulate_insn' > /sys/kernel/tracing/kprobe_events
# echo 'p:gb kvm:nonsense' > /sys/kernel/tracing/kprobe_events
bash: echo: write error: No such file or directory
# echo 'p:bg nonsense:x86_emulate_insn' > /sys/kernel/tracing/kprobe_events
# echo 'p:bb nonsense:nonsense' > /sys/kernel/tracing/kprobe_events
```

... So the only configuration that fails is `gb`, when the module is good
but the function is bad, which exactly matches the new behavior of `bpftrace`.

I think it should be good, but let me know if you think we should do additional module-checking.

Reply:
How complicated is it to check if the module is bad? Even though this mimics the sysfs behavior, I think it would be nice to give users some better messaging. Also should this succeed if the module is bad but the function is good? My gut reaction is no.

Reply:
Agreed with @jordalgo here. Even though ftrace silently succeeds with probe attachment to an invalid module, it'd be nice if we could print a warning that the module is not loaded.

Reply:
Digging a bit further, `kfunc` seems to rely on the availability of BTF for the given module/function, so fails in 3/4 cases (as it should) based on those grounds:
```
ERROR: No BTF found for kvm:nonsense
ERROR: No BTF found for nonsense:x86_emulate_insn
ERROR: No BTF found for nonsense:nonsense
```

Reply:
It looks like the way forward might be to simply parse `/proc/modules` (for example, in a function in `src/utils.cpp`), and call that before attaching a kprobe that specifies a module. Or we could just throw a warning based on that parsing.

That seems to be how the `lsmod` implementation in busybox does it:
https://github.com/brgl/busybox/blob/abbf17abccbf832365d9acf1c280369ba7d5f8b2/modutils/lsmod.c#L78

And kmod, too:
https://github.com/kmod-project/kmod/blob/48b0c75f79f86140314b9656d3d855a515e18a73/libkmod/libkmod-module.c#L1727

Reply:
Yah, I was pretty much going to suggest doing whatever `lsmod` does. So that works for me.

Reply:
Great; does that solution (a function to parse `/proc/modules`) sound good to @viktormalik too?

Reply:
@jordalgo @viktormalik My opinion is that we should mimic the `kfunc` behavior; if the module doesn't exist, bpftrace should immediately throw an error (similar to how failing in libbpf might behave).  Thoughts?

Reply:
Agreed, parsing `/proc/modules` makes sense and throwing an error if we cannot attach does, too. For wildcarded module attachpoints, we could just print a warning.